src/stmt.cpp

```diff
@@ -1870,10 +1876,12 @@ void ForeachStmt::EmitCodeForXe(FunctionEmitContext *ctx) const {
         if (sv == nullptr || ev == nullptr)
             return;
 
+        const auto svTy = startExprs[i]->GetType();
+
         // Store varying start
         sv = ctx->BroadcastValue(sv, LLVMTypes::Int32VectorType, "start_broadcast");
         llvm::Constant *delta = lCalculateDeltaForVaryingCounter(i, nDims, span);
-        sv = ctx->BinaryOperator(llvm::Instruction::Add, sv, delta, "varying_start");
```

Comment:
With these change the language becomes differently defined for Xe and not Xe targets. For Xe we generate sign and unsigned types for `foreach` counter. Is that right?

Reply:
`ForeachStmt::EmitCodeForXe` is not actually used on Xe targets. It was introduced for experiments with HW mask and can be safely removed now. BTW all code paths under `emitXeHardwareMask` can be removed.

Reply:
ok, let's remove it then as separate PR.

Reply:
See PR #2572.

Reply:
See https://github.com/ispc/ispc/pull/2572#issuecomment-1644407469. This code is unused and will be removed after PR #2572 is merged. Since the code is unused in the compiler, these changes should not block the merge.